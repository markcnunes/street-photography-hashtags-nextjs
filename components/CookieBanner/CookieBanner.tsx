import styled from '@emotion/styled';
import Link from 'next/link';
import React from 'react';
import CookieConsent from 'react-cookie-consent';
import initializeCookies, { cookieConsent } from 'utils/cookies';

const StyledCookieBanner = styled.div(({ theme }) => ({
  '.cookie': {
    position: 'fixed',
    zIndex: 100,
    left: '0',
    width: '100%',
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'center',
    alignItems: 'center',
    background: theme.colors.white,
    color: theme.colors.black,
    padding: '1.2rem 1.6rem',
    fontSize: theme.fontSizes[1],
  },

  '.cookie__message': {
    width: '100%',
    textAlign: 'center',
    margin: '0 0 1.6rem 0',
    a: {
      color: theme.colors.black,
    },
  },

  button: {
    isplay: 'inline-flex',
    alignItems: 'center',
    padding: `8px 16px`,
    borderRadius: theme.radii[2],
    border: 0,
    outline: 0,
    cursor: 'pointer',
    textDecoration: 'none',
    transition: 'background 0.3s ease',

    '&#rcc-decline-button': {
      color: theme.colors.black,
      background: theme.colors.white,
      border: '1px solid ' + theme.colors.black,
      marginRight: '1.6rem',
    },

    '&#rcc-confirm-button': {
      color: theme.colors.white,
      background: theme.colors.primary,
      marginRight: '1.6rem',
    },
  },

  '@media (min-width: 30rem)': {
    '.cookie__message': {
      width: 'auto',
      flex: '1 0 0',
      textAlign: 'left',
      margin: '0 1.6rem 0 0',
    },
  },
  '@media (min-width: 50rem)': {
    '.cookie': {
      top: 'inherit',
      bottom: '0 !important',
    },
  },
  '@media (min-width: 70rem)': {
    '.cookie__message': {
      textAlign: 'center',
      flex: 'inherit',
    },
  },
}));

const CookieBanner = () => {
  return (
    <StyledCookieBanner>
      <CookieConsent
        buttonText='Ok'
        expires={150}
        onAccept={() => {
          initializeCookies();
        }}
        enableDeclineButton
        /*
      onDecline={() => {
        alert('nay!');
      }}
      debug={true}
      */
        cookieName={cookieConsent}
        cookieValue={'true'}
        disableStyles={true}
        containerClasses='cookie'
        contentClasses='cookie__message'
      >
        This website uses cookies to analyze web traffic. To find out more, read
        our <Link href={`/privacy`}>privacy</Link> and{' '}
        <Link href={`/terms`}>terms of use</Link>.
      </CookieConsent>
    </StyledCookieBanner>
  );
};

export default CookieBanner;
