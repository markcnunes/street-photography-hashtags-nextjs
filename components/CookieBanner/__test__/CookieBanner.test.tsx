import React from 'react';

import { render } from 'test/setupTests';

import CookieBanner from '../CookieBanner';

describe('<CookieBanner />', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<CookieBanner />);
    expect(asFragment()).toMatchSnapshot();
  });
});
