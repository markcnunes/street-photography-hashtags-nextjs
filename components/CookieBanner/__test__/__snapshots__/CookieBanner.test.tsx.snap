// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CookieBanner /> matches snapshot 1`] = `
<DocumentFragment>
  .emotion-0 .cookie {
  position: fixed;
  z-index: 100;
  left: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #fff;
  color: #1c1c27;
  padding: 1.2rem 1.6rem;
  font-size: 14px;
}

.emotion-0 .cookie__message {
  width: 100%;
  text-align: center;
  margin: 0 0 1.6rem 0;
}

.emotion-0 .cookie__message a {
  color: #1c1c27;
}

.emotion-0 button {
  isplay: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 16px;
  border-radius: 4px;
  border: 0;
  outline: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
}

.emotion-0 button#rcc-decline-button {
  color: #1c1c27;
  background: #fff;
  border: 1px solid #1c1c27;
  margin-right: 1.6rem;
}

.emotion-0 button#rcc-confirm-button {
  color: #fff;
  background: #615cf5;
  margin-right: 1.6rem;
}

@media (min-width: 30rem) {
  .emotion-0 .cookie__message {
    width: auto;
    -webkit-flex: 1 0 0;
    -ms-flex: 1 0 0;
    flex: 1 0 0;
    text-align: left;
    margin: 0 1.6rem 0 0;
  }
}

@media (min-width: 50rem) {
  .emotion-0 .cookie {
    top: inherit;
    bottom: 0!important;
  }
}

@media (min-width: 70rem) {
  .emotion-0 .cookie__message {
    text-align: center;
    -webkit-flex: inherit;
    -ms-flex: inherit;
    flex: inherit;
  }
}

<div
    class="emotion-0 emotion-1"
  >
    <div
      class="cookie"
      style="bottom: 0px;"
    >
      <div
        class="cookie__message"
      >
        This website uses cookies to analyze web traffic. To find out more, read our 
        <a
          href="/privacy"
        >
          privacy
        </a>
         and 
        <a
          href="/terms"
        >
          terms of use
        </a>
        .
      </div>
      <div
        class=""
      >
        <button
          aria-label="Decline cookies"
          class=""
          id="rcc-decline-button"
        >
          I decline
        </button>
        <button
          aria-label="Accept cookies"
          class=""
          id="rcc-confirm-button"
        >
          Ok
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
